### PR TITLE
Controller branch [#158033226]

### PR DESCRIPTION
**Pull request by tjhayes**

### CODE REVIEW

File: src/ServiceHub.User.Service/Startup.cs [hunk 1]
```diff
@@ -23,13 +23,21 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddMvc();
 
-            services.AddTransient<IUserRepository, UserRepository>();
+            services.AddSingleton<IUserRepository, UserRepository>(serviceProvider =>
+            {
+                return new UserRepository(
+                    new MongoClient(connectionString)
+                    .GetDatabase("userdb")
+                    .GetCollection<Context.Models.User>("users"));
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Startup.cs]: either the mongoclient, the mongodb, or the mongocollection should be a singleton service, and the userrepo should get it that way.

File: src/ServiceHub.User.Service/Startup.cs [hunk 2]
```diff
@@ -23,13 +23,21 @@ public void ConfigureServices(IServiceCollection services)
...
-                new MongoClient(connectionString)
+            services.AddSingleton<UserRepository>(serviceProvider =>
+            {
+                return new UserRepository(
+                    new MongoClient(connectionString)
                     .GetDatabase("userdb")
-                    .GetCollection<User.Context.Models.User>("users")
-            );
+                    .GetCollection<Context.Models.User>("users"));
+            });
```
Comment (escalonn) [inline src/ServiceHub.User.Service/Startup.cs]: this second registration does nothing

File: test-dockerup.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - "9000:80"
+  db:
+    image: mongo:3.6
+    deploy:
+      replicas: 1
+    networks:
+      - appnet
+
+volumes:
+  appdata:
```
Comment (escalonn) [inline test-dockerup.yml]: why do we need a second docker compose?

Comment (tjhayes) [reply test-dockerup.yml]: it's for testing locally

**Pull request merged**